diff --git a/library/spark/spark-accumulo-library/src/main/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/rfilereaderrdd/RFileReaderIterator.java b/library/spark/spark-accumulo-library/src/main/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/rfilereaderrdd/RFileReaderIterator.java
index 82fe2c1e18..7ee91b8dea 100644
--- a/library/spark/spark-accumulo-library/src/main/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/rfilereaderrdd/RFileReaderIterator.java
+++ b/library/spark/spark-accumulo-library/src/main/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/rfilereaderrdd/RFileReaderIterator.java
@@ -103,7 +103,7 @@ public class RFileReaderIterator implements java.util.Iterator<Map.Entry<Key, Va
     private void init() throws IOException {
         final AccumuloTablet accumuloTablet = (AccumuloTablet) partition;
         LOGGER.info("Initialising RFileReaderIterator for files {}", StringUtils.join(accumuloTablet.getFiles(), ','));
-        final AccumuloConfiguration accumuloConfiguration = SiteConfiguration.getInstance(DefaultConfiguration.getInstance());
+        AccumuloConfiguration accumuloConfiguration = SiteConfiguration.getInstance();
 
         // Required column families according to the configuration
         final Set<ByteSequence> requiredColumnFamilies = InputConfigurator
@@ -129,7 +129,7 @@ public class RFileReaderIterator implements java.util.Iterator<Map.Entry<Key, Va
         // Apply visibility filtering iterator
         if (null != auths) {
             final Authorizations authorizations = new Authorizations(auths.toArray(new String[auths.size()]));
-            final VisibilityFilter visibilityFilter = new VisibilityFilter(mergedIterator, authorizations, new byte[]{});
+            final SortedKeyValueIterator<Key, Value> visibilityFilter =  VisibilityFilter.wrap(mergedIterator, authorizations, new byte[]{});
             final IteratorSetting visibilityIteratorSetting = new IteratorSetting(1, "auth", VisibilityFilter.class);
             visibilityFilter.init(mergedIterator, visibilityIteratorSetting.getOptions(), null);
             iteratorAfterIterators = visibilityFilter;
